src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FileMigrationServiceTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Common.Tests.Utils.MockUtils;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.Tests.Utils.ContentDbUtils;
+using static GovUk.Education.ExploreEducationStatistics.Publisher.Model.PublisherQueues;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services;
+
+public class FileMigrationServiceTests
+{
+    [Fact]
+    public async Task MigrateAll_FailsWithNullMessageCount()
+    {
+        var storageQueueService = MockStorageQueueService(messageCount: null);
+
+        var service = SetupService(
+            contentDbContext: InMemoryContentDbContext(),
+            storageQueueService: storageQueueService.Object);
+
+        var result = await service.MigrateAll();
+
+        VerifyAllMocks(storageQueueService);
+
+        var left = result.AssertLeft();
+        Assert.IsAssignableFrom<BadRequestObjectResult>(left);
+    }
+
+    [Fact]
+    public async Task MigrateAll_FailsWithNonZeroMessageCount()
+    {
+        var storageQueueService = MockStorageQueueService(messageCount: 1);
+
+        var service = SetupService(
+            contentDbContext: InMemoryContentDbContext(),
+            storageQueueService: storageQueueService.Object);
+
+        var result = await service.MigrateAll();
+
+        VerifyAllMocks(storageQueueService);
+
+        var left = result.AssertLeft();
+        Assert.IsAssignableFrom<BadRequestObjectResult>(left);
+    }
+
+    [Fact]
+    public async Task MigrateAll_IgnoresFilesWithContentTypeOrSize()
+    {
+        // Set up files that have content type, size, or both content type and size
+        var files = new List<File>
+        {
+            new()
+            {
+                Id = Guid.NewGuid(),
```

Comment:
But do we not want to be able to fix up Files that have been partially migrated upon a rerun?  E.g. I'd assume that a File with only ContentType but no Size isn't valid, so wouldn't we want to revisit those on a subsequent run?